src/graph/validator/GoValidator.cpp

```diff
@@ -210,10 +301,55 @@ Status GoValidator::buildColumns() {
     goCtx_->filter = rewrite2VarProp(newFilter);
   }
 
+  bool existEdge = false;
   auto* newYieldExpr = pool->add(new YieldColumns());
-  for (auto* yield : goCtx_->yieldExpr->columns()) {
-    extractPropExprs(yield->expr());
-    newYieldExpr->addColumn(new YieldColumn(rewrite2VarProp(yield->expr()), yield->alias()));
+  for (auto* col : goCtx_->yieldExpr->columns()) {
+    if (ExpressionUtils::hasAny(col->expr(),
+                                {Expression::Kind::kVertex, Expression::Kind::kEdge})) {
+      std::string colName = col->name();
+      if (colTypeMap_.find(colName) != colTypeMap_.end() && colTypeMap_[colName]) {
+        goCtx_->srcEdgePropsExpr->addColumn(
+            new YieldColumn(VertexExpression::make(pool), COLNAME_SRC_VERTEX));
+        if (col->expr()->kind() == Expression::Kind::kVertex) {
+          col->setExpr(rewriteVertex2VarProp(col->expr()));
+          newYieldExpr->addColumn(
+              new YieldColumn(InputPropertyExpression::make(pool, COLNAME_SRC_VERTEX), colName));
+        } else {
+          col->setExpr(rewriteVertex2VarProp(col->expr()));
+          newYieldExpr->addColumn(col->clone().release());
+        }
+      }
+      if (colTypeMap_.find(colName) != colTypeMap_.end() && !colTypeMap_[colName]) {
+        goCtx_->dstPropsExpr->addColumn(
+            new YieldColumn(VertexExpression::make(pool), COLNAME_DST_VERTEX));
+        if (col->expr()->kind() == Expression::Kind::kVertex) {
+          col->setExpr(rewriteVertex2VarProp(col->expr()));
+          newYieldExpr->addColumn(
+              new YieldColumn(InputPropertyExpression::make(pool, COLNAME_DST_VERTEX), colName));
+        } else {
+          col->setExpr(rewriteVertex2VarProp(col->expr()));
+          newYieldExpr->addColumn(col->clone().release());
+        }
+      }
+      if (ExpressionUtils::hasAny(col->expr(), {Expression::Kind::kEdge})) {
+        if (!existEdge) {
+          goCtx_->srcEdgePropsExpr->addColumn(
+              new YieldColumn(EdgeExpression::make(pool), COLNAME_EDGE));
+          existEdge = true;
+        }
+        if (col->expr()->kind() == Expression::Kind::kEdge) {
+          col->setExpr(rewriteEdge2VarProp(col->expr()));
+          newYieldExpr->addColumn(
+              new YieldColumn(InputPropertyExpression::make(pool, COLNAME_EDGE), colName));
```

Comment:
Maybe we need not to rewrite `VetexExpression` into `InputPropertyExpression` since we can modify the `ExpressionContext::getVertex(const std::string& name)` interface to implement your function.

Reply:
done